### PR TITLE
feat: add `foreach` snippet to `CompletionProvider`

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to #4224

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala [hunk 1]
```diff
@@ -236,6 +237,12 @@ object CompletionProvider {
   private def getSnippetCompletions()(implicit context: Context, index: Index, root: TypedAst.Root): List[CompletionItem] = {
     List(
       // NB: Please keep the list alphabetically sorted.
+      snippetCompletion("foreach",
+        s"""foreach ($${1:pat} <- $${2:iterable}) {
+           |    $${3:body}
+           |}""".stripMargin,
```
Comment (jaschdoc) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala:243]: Should we add a final semicolon at the end of the block scope?

**Comment by jaschdoc** [comment]
RFR

**Comment by magnus-madsen** [comment]
Sorry. I said snippet but actually just meant the keyword. We don't use snippets anymore (except for main).

**Comment by jaschdoc** [comment]
Alright, no worres

**Pull request merged**